_inc/lib/plugins.php

```diff
@@ -0,0 +1,173 @@
+<?php
+/**
+ * Plugins Library
+ *
+ * Helper functions for installing and activating plugins.
+ *
+ * Used by the REST API
+ *
+ * @autounit api plugins
+ */
+
+include_once ABSPATH . 'wp-admin/includes/class-wp-upgrader.php';
+include_once ABSPATH . 'wp-admin/includes/file.php';
+
+class Jetpack_Plugins {
+
+	 /**
+	 * Install a plugin.
+	 *
+	 * @since 5.8.0
+	 *
+	 * @param string $slug Plugin slug.
+	 *
+	 * @return bool|WP_Error True if installation succeeded, error object otherwise.
+	 */
+	public static function install_plugin( $slug ) {
+		if ( is_multisite() && ! current_user_can( 'manage_network' ) ) {
+			return new WP_Error( 'not_allowed', 'You are not allowed to install plugins on this site.' );
+		}
+
+		$skin     = new Jetpack_Automatic_Install_Skin();
+		$upgrader = new Plugin_Upgrader( $skin );
+		$zip_url  = self::generate_wordpress_org_plugin_download_link( $slug );
+
+		return $upgrader->install( $zip_url );
+	}
+
+	 protected static function generate_wordpress_org_plugin_download_link( $plugin_slug ) {
+		return "https://downloads.wordpress.org/plugin/{$plugin_slug}.latest-stable.zip";
+	 }
+
+	 public static function get_plugin_id_by_slug( $slug ) {
+		/** This filter is documented in wp-admin/includes/class-wp-plugins-list-table.php */
+		$plugins = apply_filters( 'all_plugins', get_plugins() );
+		if ( ! is_array( $plugins ) ) {
+			return false;
+		}
+		foreach ( $plugins as $plugin_file => $plugin_data ) {
+			if ( self::get_slug_from_file_path( $plugin_file ) === $slug ) {
+				return $plugin_file;
+			}
+		}
+
+		return false;
+	}
+
+	protected static function get_slug_from_file_path( $plugin_file ) {
+		// Simular to get_plugin_slug() method.
```

Comment:
Should be "Simular".

Reply:
You folks both probably meant "Similar" 😉

Reply:
☝️